feat(*): add support for test-failure hook

michelleN opened the pull request:
* resolves #1927 

Can only be merged after rebasing on master with #1935

Commit:
feat(helm): add cleanup flag to test command

Commit:
feat(*): add support for test-failure hook

* resolved #1927

Commit:
ref(pkg/releasetesting): ref environment tests

* take out interfaces since no longer needed

vdice commented on pkg/releasetesting/test_suite.go:
```diff
@@ -61,89 +61,97 @@ func NewTestSuite(rel *release.Release) (*TestSuite, error) {
+			return false, nil
+		}
+	}
+	return false, errors.New("No test-success or test-failure hook found")
```
maybe format in actual hook consts in case hook values change in future? `errors.New(fmt.Sprintf("No %s or %s hook found", hooks.ReleaseTestSuccess, hooks.ReleaseTestFailure))`

Alternatively, perhaps create a specific `ErrUnknownTestHook` error that can be thrown?

vdice commented on pkg/releasetesting/test_suite_test.go:
```diff
@@ -37,6 +37,29 @@ import (
 	tillerEnv "k8s.io/helm/pkg/tiller/environment"
 )
 
+const manifestWithTestHook = `
```
should we add a manifest for the new `test-failure` hook and associated test(s)?

michelleN replied:
yes

technosophos commented:
If this needs to make it into 2.2.0, please let me know in Slack. Otherwise I will assume this is a 2.3.0 feature.

michelleN commented on pkg/releasetesting/test_suite.go:
```diff
@@ -61,89 +61,97 @@ func NewTestSuite(rel *release.Release) (*TestSuite, error) {
+		}
+	case api.PodFailed:
+		if t.expectedSuccess == false {
+			fmt.Println("expected to be here")
```
nope. thanks

michelleN commented:
I am closing this PR in favor of #1939 -- I had rebased on an earlier PR which changed and caused a bunch of issues. @vdice - I addressed all of your comments in the new PR. thank you.
